### PR TITLE
use xdg to store rho related config and data files. Closes #136.

### LINKED ISSUE #136: Store profiles and auths in consistent location
Right now, profiles and auths are stored in a `data` directory inside the current working directory of the user who runs rho. If the user changes their working directory, they lose their profile and auth information.

Based on http://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/var.html, either `/var/local` or `/var/opt` would be appropriate places, depending on whether we decide to package rho inside `/usr/local` or `/opt`.

### PR DESCRIPTION
**Pull request by chambridge**

### CODE REVIEW

File: rho/utilities.py [hunk 1]
```diff
@@ -18,16 +18,24 @@
...
 else:
     import thread as _thread  # pylint: disable=import-error
 
-
-CREDENTIALS_PATH = 'data/credentials'
-PROFILES_PATH = 'data/profiles'
+RHO_PATH = 'rho'
+CONFIG_DIR = os.path.join(xdg_config_home, RHO_PATH)
+DATA_DIR = os.path.join(xdg_data_home, RHO_PATH)
+RHO_LOG = os.path.join(CONFIG_DIR, 'rho_log')
```
Comment (noahl) [inline rho/utilities.py]: I think `RHO_LOG` makes more sense in `DATA_DIR`, not `CONFIG_DIR`

File: README.rst [hunk 2]
```diff
@@ -62,12 +62,13 @@ a CLI prompt.
 (The different rho commands are covered more in the `Command Syntax & Usage`_
 section.)
 
-All the information that rho needs is stored in the data folder in
-the installed directory. All the auths are stored in the ``credentials``
-file. All the profiles are stored in the ``profiles`` file. The Ansible
-playbook is called ``rho_playbook.yml`` stored in the installed directory.
-The roles created during the scan are stored in the ``roles`` folder. These
-roles are used by ``rho_playbook.yml`` to perform the fact collection.
+All the information that rho needs is stored in the XDG_CONFIG_HOME/rho and
```
Comment (noahl) [inline README.rst]: I've seen these directories written as `$XDG_CONFIG_HOME/rho` (with the leading `$`) before. I think the version with the leading `$` might be more idiomatic, but have you seen it differently?

**Review by noahl — APPROVED** [review]
Two small comments, but this looks great!

**Review by elyezer — APPROVED** [review]
Looks good to me.

File: rho/utilities.py (hunk 1 above)
Comment (chambridge) [reply rho/utilities.py]: I'll update this. Thanks for the feedback.

File: README.rst (hunk 2 above)
Comment (chambridge) [reply README.rst]: updated with the leading `$`

**Pull request merged**